src/data/Scenario.h

```diff
@@ -15,7 +15,9 @@
 // they are set as static constant values in case they need to be shared with
 // other classes (e.g. solvers)
 //
-
+enum PREF_LEVEL {WEAK=0, MODERATE=1, STRONG=2, COMPULSORY=3};
```

Comment:
I am really not a big fan of hard coded levels of preferences. It would really be better to have it depend on input parameters.